web/packages/teleport/src/JoinTokens/JoinTokenForms.tsx

```diff
@@ -0,0 +1,224 @@
+            <Text fontWeight={700} mb={2}>
+              AWS Rule
+            </Text>
+
+            {rules.length > 1 && ( // at least one rule is required, so lets not allow the user to remove it
+              <ButtonIcon onClick={() => removeRule(index)}>
+                <Trash size={16} color="text.muted" />
+              </ButtonIcon>
+            )}
+          </Flex>
+          <FieldInput
+            label="Account ID"
+            rule={requiredField('Account ID is required')}
+            placeholder="AWS Account ID"
+            value={rule.aws_account}
+            onChange={e =>
+              setTokenRulesField(index, 'aws_account', e.target.value)
+            }
+          />
+          <FieldInput
+            label="ARN"
+            toolTipContent={`The joining nodes must match this ARN. Supports wildcards "*" and "?"`}
+            placeholder="arn:aws:iam::account-id:role/*"
+            value={rule.aws_arn}
+            onChange={e => setTokenRulesField(index, 'aws_arn', e.target.value)}
+          />
+        </RuleBox>
+      ))}
+      <ButtonText onClick={addNewRule}>
+        <Plus size={16} mr={2} />
+        Add another AWS Rule
+      </ButtonText>
+    </>
+  );
+};
+
+export const JoinTokenGCPForm = ({
+  tokenState,
+  onUpdateState,
+}: {
+  tokenState: NewJoinTokenState;
+  onUpdateState: (newToken: NewJoinTokenState) => void;
+}) => {
+  const rules = tokenState.gcp;
+  function removeRule(index: number) {
+    const newRules = rules.filter((_, i) => index !== i);
+    const newState = {
+      ...tokenState,
+      gcp: newRules,
+    };
+    onUpdateState(newState);
+  }
+
+  function addNewRule() {
+    const newState = {
+      ...tokenState,
+      gcp: [
+        ...tokenState.gcp,
+        { project_ids: [], locations: [], service_accounts: [] },
+      ],
+    };
+    onUpdateState(newState);
+  }
+
+  function updateRuleField(
+    index: number,
+    fieldName: string,
+    opts: OptionGCP[]
+  ) {
+    const newState = {
+      ...tokenState,
+      gcp: tokenState.gcp.map((rule, i) => {
+        if (i === index) {
+          return { ...rule, [fieldName]: opts };
+        }
+        return rule;
+      }),
+    };
+    onUpdateState(newState);
+  }
+
+  return (
+    <>
+      {rules.map((rule, index) => (
+        <RuleBox
+          key={index} // order doesn't change without updating the referrenced array
+        >
+          <Flex alignItems="center" justifyContent="space-between">
+            <Text fontWeight={700} mb={2}>
+              GCP Rule
+            </Text>
+
+            {rules.length > 1 && ( // at least one rule is required, so lets not allow the user to remove it
+              <ButtonIcon onClick={() => removeRule(index)}>
+                <Trash size={16} color="text.muted" />
+              </ButtonIcon>
+            )}
+          </Flex>
+          <FieldSelectCreatable
+            placeholder="Type a project ID"
```

Comment:
We should be consistent on capitalisation (`project ID` vs `Project ID` in the required rule)